src/extensions/lens-extension.ts

```diff
@@ -14,35 +13,17 @@ export interface LensExtensionManifest {
   renderer?: string; // path to %ext/dist/renderer.js
 }
 
-export interface LensExtensionStoreModel {
-  isEnabled: boolean;
-}
-
-export class LensExtension<S extends ExtensionStore<LensExtensionStoreModel> = any> {
-  protected store: S;
+export class LensExtension {
   readonly manifest: LensExtensionManifest;
   readonly manifestPath: string;
   readonly isBundled: boolean;
 
+  @observable private isEnabled = false;
```

Comment:
With this being private, I don't think it implements the `InstalledExtension` interface anymore, is that desirable?

Reply:
Why you think it should implement? Using `private` field was suggested by @jakolehm

Reply:
Maybe I don't have a grasp of how these types interact, I would have assumed that the `extension-manager.ts` file keeps a record to instances of this type. Does it not?

Reply:
Nope.. `extension-loader.ts` keeps both `InstalledExtension` list and enabled instances of those.

Reply:
IMO that seems very strange, because we are keeping two lists of information that should always be the same. Why not just one list?

Reply:
Extension-manager doesn't keep anything.. Its purpose for install & load extensions.

Reply:
Isn't `extension-loader` for installing and loading extensions? I would have assumed that `extension-manager` is the manager of truth and then uses `loader` to load stuff and `store` to persist data between restarts...

Reply:
About private: without private extension instance itself cold toggle this which does not make any sense to me and is not really desirable. Why would we allow extension to do this?

`ExtensionLoader`: it's only purpose is to load (read: execute) extension instances. List of extensions are given to this class, it does not care how that list has generated.

`ExtensionManager`: it's only purpose is to gather a list of extensions and then _install_ those to central `node_modules`  folder (including all dependencies). Note: this does not _load_ (or create) extension instances.

Naming things is one of the most difficult things in computer science so we probably could have better names for these. Still I strongly believe that separating complex things to smaller units is better than having one giant class that tries to do everything.